### ISSUE TITLE
Unable to build node.js typescript project using the library: Missing exported members

### ISSUE THREAD

**Issue by JonathanIlk** [freshgum-bubbles/typedi#145]
**Describe the bug**
Project does not build, due to typedi build errors:

```
> node-typescript-boilerplate@0.0.0 build
> tsc -p tsconfig.json                   

node_modules/@freshgum/typedi/build/esm5/container-instance.class.d.mts:4:58 - error TS2724: '"./interfaces/service-options.interface.mjs"' has no exported member named 'ServiceOptionsWithoutTypeOrDependencies'. Did you mean 'ServiceOptionsWithoutDependencies'?
                                                                                                                                                                                                                                                                     
4 import { ServiceOptions, ServiceOptionsWithDependencies, ServiceOptionsWithoutTypeOrDependencies } from './interfaces/service-options.interface.mjs';                                                                                                              
                                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                   

node_modules/@freshgum/typedi/build/esm5/container-instance.class.d.mts:12:10 - error TS2305: Module '"./types/service-identifier-location.type.mjs"' has no exported member 'ServiceIdentifierLocation'.

12 import { ServiceIdentifierLocation } from './types/service-identifier-location.type.mjs';
            ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@freshgum/typedi/build/esm5/functions/lazy.function.d.mts:2:10 - error TS2305: Module '"../types/infer-service-type.type.mjs"' has no exported member 'InferServiceType'.

2 import { InferServiceType } from '../types/infer-service-type.type.mjs';
           ~~~~~~~~~~~~~~~~

node_modules/@freshgum/typedi/build/esm5/index.d.mts:21:10 - error TS2305: Module '"./types/service-identifier-location.type.mjs"' has no exported member 'ServiceIdentifierLocation'.

21 export { ServiceIdentifierLocation } from './types/service-identifier-location.type.mjs';
            ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@freshgum/typedi/build/esm5/interfaces/tree-visitor.interface.d.mts:4:10 - error TS2305: Module '"../types/service-identifier-location.type.mjs"' has no exported member 'ServiceIdentifierLocation'.

4 import { ServiceIdentifierLocation } from '../types/service-identifier-location.type.mjs';
           ~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@freshgum/typedi/build/esm5/types/type-wrapper.type.d.mts:3:10 - error TS2305: Module '"./infer-service-type.type.mjs"' has no exported member 'InferServiceType'.

3 import { InferServiceType } from './infer-service-type.type.mjs';
           ~~~~~~~~~~~~~~~~


Found 6 errors in 5 files.

Errors  Files
     2  node_modules/@freshgum/typedi/build/esm5/container-instance.class.d.mts:4
     1  node_modules/@freshgum/typedi/build/esm5/functions/lazy.function.d.mts:2
     1  node_modules/@freshgum/typedi/build/esm5/index.d.mts:21
     1  node_modules/@freshgum/typedi/build/esm5/interfaces/tree-visitor.interface.d.mts:4
     1  node_modules/@freshgum/typedi/build/esm5/types/type-wrapper.type.d.mts:3
```


**To Reproduce**

Use preconfigured project:
1. Clone https://github.com/JonathanIlk/typedi-build-issue
2. `npm run build`

DIY:
1. Download typescript node.js boilerplate project (https://github.com/jsynowiec/node-typescript-boilerplate)
2. Add @freshgum/typedi as dependency to package.json
3. Add experimentalDecorators to tsconfig.json
4. Add class with @Service Decorator to main.ts
5. `npm run build`

**Expected behavior**
Project builds correctly.

**Desktop (please complete the following information):**

- OS: Windows
- Typescript: 5.1


**Additional context**
The missing types appear to be missing from the interface files. However they seem to be available in the source code in the github.
Maybe they went missing when deploying the package to npm?

**Comment by freshgum-bubbles** [comment]
Thanks for the report! Looking into it now.  That shouldn't happen; all the types should be there, but apparently not :-S

**Comment by freshgum-bubbles** [comment]
Hi, figured out the issue: due to us using `NodeNext` for `moduleResolution`, you have to use it in your package's tsconfig too.  Currently investigating alternatives (as this seems a bit dumb).  There was a related issue regarding internal types (#146) which should be fixed with 0.7.1 -- can you try it and report back?

**Comment by JonathanIlk** [comment]
Works now with 0.7.1 👍
